### ISSUE TITLE
Wala Error while creating CHA call-graph

### ISSUE THREAD

**Issue by akshayutture** [wala/WALA#890]
I was running WALA on the NJR benchmarks (https://zenodo.org/record/4632231#.YL_FGi9h2C) 
and a small fraction of the benchmarks (9 out of 293) run into
the following error when creating a CHA call-graph.

Added below is the code used and error message.
Here is the link to one such benchmark:
https://drive.google.com/file/d/1qUt2mpiR2AnQFyiCkSM-bzMF7JM13SqP/view?usp=sharing

### Code
```
import java.io.IOException;
import java.util.Properties;
import com.ibm.wala.ipa.callgraph.AnalysisScope;
import com.ibm.wala.ipa.callgraph.Entrypoint;
import com.ibm.wala.ipa.callgraph.cha.CHACallGraph;
import com.ibm.wala.ipa.callgraph.impl.Util;
import com.ibm.wala.ipa.cha.ClassHierarchyException;
import com.ibm.wala.ipa.cha.ClassHierarchyFactory;
import com.ibm.wala.ipa.cha.IClassHierarchy;
import com.ibm.wala.util.CancelException;
import com.ibm.wala.util.config.AnalysisScopeReader;
import com.ibm.wala.util.io.CommandLine;

public class WalaError {
	public static void main(String[] args) throws ClassHierarchyException, IOException, CancelException{
		Properties p = CommandLine.parse(args);
		String classpath = p.getProperty("jarfile");
		
		AnalysisScope scope = AnalysisScopeReader.makeJavaBinaryAnalysisScope(classpath, null);
		IClassHierarchy cha = ClassHierarchyFactory.make(scope);
		Iterable<Entrypoint> entrypoints = Util.makeMainEntrypoints(scope, cha);
		CHACallGraph cg = new CHACallGraph(cha);
		cg.init(entrypoints);
	}
}
```

### Command
java -cp <wala_jars> WalaError -jarfile <jarfile_name>

### Error Message
Fatal error at index 616
Fatal error at index 581
Fatal error at index 566
Fatal error at index 38
Exception in thread "main" java.lang.IllegalArgumentException: Constant pool item #-32015 out of range
	at com.ibm.wala.shrikeCT.ConstantPoolParser.getItemType(ConstantPoolParser.java:161)
	at com.ibm.wala.shrikeBT.shrikeCT.CTDecoder$CPReader.getConstantPoolItemType(CTDecoder.java:78)
	at com.ibm.wala.shrikeBT.ConstantInstruction.make(ConstantInstruction.java:755)
	at com.ibm.wala.shrikeBT.Decoder.makeConstantPoolLoad(Decoder.java:207)
	at com.ibm.wala.shrikeBT.Decoder.decodeBytecodeInstruction(Decoder.java:505)
	at com.ibm.wala.shrikeBT.Decoder.decodeAt(Decoder.java:807)
	at com.ibm.wala.shrikeBT.Decoder.decodeAt(Decoder.java:827)
	at com.ibm.wala.shrikeBT.Decoder.decodeAt(Decoder.java:827)
	at com.ibm.wala.shrikeBT.Decoder.decodeAt(Decoder.java:827)
	at com.ibm.wala.shrikeBT.Decoder.decode(Decoder.java:1014)
	at com.ibm.wala.classLoader.ShrikeCTMethod.makeDecoder(ShrikeCTMethod.java:103)
	at com.ibm.wala.classLoader.ShrikeBTMethod.processBytecodesWithShrikeBT(ShrikeBTMethod.java:417)
	at com.ibm.wala.classLoader.ShrikeBTMethod.computeBCInfo(ShrikeBTMethod.java:223)
	at com.ibm.wala.classLoader.ShrikeBTMethod.getBCInfo(ShrikeBTMethod.java:129)
	at com.ibm.wala.classLoader.ShrikeBTMethod.getCallSites(ShrikeBTMethod.java:181)
	at com.ibm.wala.classLoader.CodeScanner.getCallSitesFromShrikeBT(CodeScanner.java:186)
	at com.ibm.wala.classLoader.CodeScanner.getCallSites(CodeScanner.java:49)
	at com.ibm.wala.ipa.callgraph.cha.ContextInsensitiveCHAContextInterpreter.iterateCallSites(ContextInsensitiveCHAContextInterpreter.java:34)
	at com.ibm.wala.ipa.callgraph.cha.CHACallGraph$CHANode.iterateCallSites(CHACallGraph.java:87)
	at com.ibm.wala.ipa.callgraph.cha.CHACallGraph.closure(CHACallGraph.java:249)
	at com.ibm.wala.ipa.callgraph.cha.CHACallGraph.init(CHACallGraph.java:128)
	at main.npa.WalaError.main(WalaError.java:25)_

**Comment by akshayutture** [comment]
The jar file was compiled with Java8. Here are the class file version numbers:
minor version: 0
major version: 52

**Comment by msridhar** [comment]
Thanks again for the report @akshayutture!  This should now be fixed on master.  Let us know if you need us to cut a new release with the fix.

**Comment by akshayutture** [comment]
Thank you so much for the prompt fix! 
Yes, it would be really helpful if you could cut a new release with the fix.

**Comment by msridhar** [comment]
@akshayutture before I cut a new release, can you test that the latest code doesn't crash for your data set?  You can test by using WALA version `1.5.7-SNAPSHOT`, either by adding the Sonatype snapshots repo to your build, https://oss.sonatype.org/content/repositories/snapshots/, or by installing to your local Maven repo by cloning and then running `./gradlew publishToMavenLocal`.

**Comment by akshayutture** [comment]
Sure. I shall run it on our data-set and get back to you with the results (about crashes) in a couple days.

**Comment by akshayutture** [comment]
I was actually just about to start the run, but I was unable to access the snapshot at oss.sonatype. The link you sent me gives me a "Directory listing forbidden" error. Is there any other way for me to access it?

**Comment by msridhar** [comment]
@akshayutture sorry I didn't know you would be downloading the jars directly.  Here is the root directory that is browsable:

https://oss.sonatype.org/content/repositories/snapshots/com/ibm/wala/

You'll want the most recent `1.5.7-SNAPSHOT` jars for `com.ibm.wala.core`, `com.ibm.wala.shrike`, and `com.ibm.wala.util` I think.  Alternately, you can run `./gradlew publishToMavenLocal` after cloning the master branch, and then these jars will be available under your `~/.m2/repository` directory.

**Comment by akshayutture** [comment]
@msridhar I have run the 1.5.7-SNAPSHOT from June12 on our dataset, and there were no errors for the CHA-Callgraph construction.
Further, there were no errors for the 0-CFA call-graph construction with a 2 minute time-limit on each benchmark. 
(I didn't run 0-CFA until it terminated because the benchmarks are really large and the run takes weeks)

**Comment by msridhar** [comment]
@akshayutture I was cutting a release just now but ran into a problem (#898).  Once that gets fixed properly, I will cut a new release.

**Comment by msridhar** [comment]
@akshayutture finally we have fixed our build issues, and WALA 1.5.7 is released with this fix

**Comment by akshayutture** [comment]
That's great! Thank you so much. It's great that we can now run our experiments with 100% of our benchmarks working.
